app/plugin_install.go

```diff
@@ -31,6 +53,30 @@ func (a *App) InstallPlugin(pluginFile io.ReadSeeker, replace bool) (*model.Mani
 }
 
 func (a *App) installPlugin(pluginFile io.ReadSeeker, replace bool) (*model.Manifest, *model.AppError) {
+	manifest, appErr := a.installPluginLocally(pluginFile, replace)
+	if appErr != nil {
+		return nil, appErr
+	}
+
+	// Store bundle in the file store to allow access from other servers.
+	pluginFile.Seek(0, 0)
+
+	if _, err := a.WriteFile(pluginFile, a.getBundleStorePath(manifest.Id)); err != nil {
+		return nil, model.NewAppError("uploadPlugin", "app.plugin.store_bundle.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	a.notifyClusterPluginEvent(
+		model.CLUSTER_EVENT_INSTALL_PLUGIN,
+		model.PluginEventData{
+			Id:            manifest.Id,
+			FileStorePath: a.getBundleStorePath(manifest.Id),
```

Comment:
Why bother passing the filestorepath? Shouldn't the other server just derive it from the manifest.Id?

Reply:
This change comes from the related [PR]( https://github.com/mattermost/mattermost-server/pull/11611) @ali-farooq0

Reply:
@crspeller the thinking there was to not let the server figure out the path and just use what was passed through. However since we're figuring out the path anyway, no point is passing it through. I'll make the change in the feature branch before merging it into master.